### PR TITLE
Makefile written in GO

### PR DESCRIPTION
**Pull request by ovisan**
this is the initial commit that has ```install``` and ```all``` commands working
https://github.com/mesosphere/kubernetes-mesos/issues/191

### CODE REVIEW

File: makefile.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	target          = "./target"
+)
+
+func run(dir, command string, args ...string) (output string, err error) {
+	cmd := exec.Command(command, args...)
+	cmd.Dir = dir
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
```
Comment (jdef) [inline makefile.go]: why capture the output/error streams instead of just configuring cmd to use `os.Stdout` and `os.Stdin`?

File: makefile.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	KUBE_GO_PACKAGE string
+	K8SM_GO_PACKAGE string
+	K8_CMD          []string
+	tempDirs        []string
+	CMD_DIRS        []string
+	LIB_DIRS        []string
+	target          = "./target"
+)
+
+func run(dir, command string, args ...string) (output string, err error) {
```
Comment (jdef) [inline makefile.go]: the output string is never actually used anywhere, should probably kill it and just return error

File: makefile.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	"os/exec"
+	"path"
+	"path/filepath"
+)
+
+var (
+	KUBE_GO_PACKAGE string
+	K8SM_GO_PACKAGE string
+	K8_CMD          []string
+	tempDirs        []string
```
Comment (jdef) [inline makefile.go]: smells funny to have this as a global

**Comment by jdef** [comment]
Thanks for the PR. This is a nice first step towards breaking away from the Makefile. I'd recommend adding something like `func dieIf(err error) { if err != nil { log.Fatal(err) } }` that you can wrap around calls that simply return `error` -- saves on some of the boilerplate.

**Comment by ConnorDoyle** [comment]
Could you also make the associated changes to the build instructions in the README and elsewhere?

File: makefile.go (hunk 1 above)
Comment (ovisan) [inline makefile.go]: I did it like this because later on it's maybe needed to parse the output for further actions, but it can be changed back later.

**Comment by ovisan** [comment]
You are welcome :-) 
The README contains now the build instructions and the code has been modified with your suggestions.

**Pull request closed without merging**